ProcessLib/GroundwaterFlow/GroundwaterFlowFEM.h

```diff
@@ -64,29 +65,29 @@ class LocalAssemblerData
     /// The hydraulic_conductivity factor is directly integrated into the local
     /// element matrix.
     LocalAssemblerData(MeshLib::Element const& element,
-                       std::size_t const local_matrix_size,
+                       std::size_t const /*local_matrix_size*/,
                        unsigned const integration_order,
                        GroundwaterFlowProcessData const& process_data)
         : _element(element)
         , _shape_matrices(
               initShapeMatrices<ShapeFunction, ShapeMatricesType, IntegrationMethod, GlobalDim>(
                   element, integration_order))
         , _process_data(process_data)
-        , _localA(local_matrix_size, local_matrix_size) // TODO narrowing conversion
-        , _localRhs(local_matrix_size)
         , _integration_order(integration_order)
     {
-        // This assertion is valid only if all nodal d.o.f. use the same shape matrices.
-        assert(local_matrix_size == ShapeFunction::NPOINTS * NUM_NODAL_DOF);
     }
 
-    void assembleConcrete(
-        double const t, std::vector<double> const& local_x,
-        NumLib::LocalToGlobalIndexMap::RowColumnIndices const& indices,
-        GlobalMatrix& /*M*/, GlobalMatrix& K, GlobalVector& b) override
+    void assemble(double const t, std::vector<double> const& local_x,
+                  std::vector<double>& /*local_M_data*/,
+                  std::vector<double>& local_K_data,
+                  std::vector<double>& /*local_b_data*/) override
```

Comment:
I forgot your explanation on why these parameters are all std::vecs and not GlobalMatrix/Vector anymore. For discussion...